clientv3/watch.go

```diff
@@ -283,7 +288,10 @@ func (w *watcher) Watch(ctx context.Context, key string, opts ...OpOption) Watch
 	// receive channel
 	if ok {
 		select {
-		case ret := <-retc:
+		case ret, ok := <-wr.retc:
```

Comment:
can we add some comments for this? it is not clear to me why do we need to call watch recursively.